presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -97,7 +97,7 @@ public double overlapPercentWith(StatisticRange other)
         double lengthOfIntersect = min(high, other.high) - max(low, other.low);
         if (isInfinite(lengthOfIntersect)) {
             if (isFinite(this.distinctValues) && isFinite(other.distinctValues)) {
-                return min(this.distinctValues, other.distinctValues) / max(this.distinctValues, other.distinctValues);
+                return min(other.distinctValues / this.distinctValues, 1);
```

Comment:
The function estimates in what percentage the range `this` overlaps with range `other`. It must not return anything greater that 1.
This branch is used when we cannot use ranges lengths, but have NDVs at hand.
I do not recall why picked this heuristic (which query was misbehaving).
@findepi ?